### DISCUSSION TITLE
Bounce records data in webhooks

### DISCUSSION THREAD

**Discussion by kayastha1988** [postalserver/postal#2467] [Help with using Postal]
After running email campaign for 3rd party service, I have result of all the sent emails. How to get all those email information data in webhooks? Couldn't find proper documentation on this. 
Thanks

**Comment by willpower232** [comment]
Webhooks send messages back to your application when the event occurs so if you haven't already set them up, you won't be able to receive events that have already happened unfortunately.

You will need to have some code that understands webhooks and then you can enter the URL of the code into the Postal UI when setting up the webhook. Have a look at https://docs.postalserver.io/developer/webhooks

**Reply by kayastha1988** [reply]
Would you mind providing me that code so that I can get those email data from webhooks? In that URL, just json output is shown but not said how to get these data in URL that we define.

Can you help me please?

**Reply by willpower232** [reply]
webhooks just tell you what happened to a message, if you aren't recording the message ID when sending with the API then you should start doing that so you can keep a copy of the message and know how Postal refers to it.

You could retrieve more information from the API once you have the message ID from the webhook if you wanted https://apiv1.postalserver.io/controllers/messages/message.html
